src/BLE_Server.cpp

```diff
@@ -142,13 +142,13 @@ void computeERG(int currentWatts, int setPoint) {
 
   if (cad > 20) {
     // 30  is amount of watts per shift. Was 50, seemed like too much...
-    if (abs(currentWatts - setPoint) < 30) {
-      amountToChangeIncline = (currentWatts - setPoint) * 1;
+    if (abs(currentWatts - setPoint) < WATTS_PER_SHIFT) {
+      amountToChangeIncline = (currentWatts - setPoint) * userConfig.getERGSensitivity();
     }
-    if (abs(currentWatts - setPoint) > 30) {
-      amountToChangeIncline = amountToChangeIncline + ((currentWatts - setPoint)) * 1;
+    if (abs(currentWatts - setPoint) > WATTS_PER_SHIFT) {
+      amountToChangeIncline = amountToChangeIncline + ((currentWatts - setPoint)) * userConfig.getERGSensitivity();
```

Comment:
At this point `amountToChange` is `0` so whether the watt change is greater or less than `WATTS_PER_SHIFT`, `amountToChangeIncline` is the same.

Reply:
Good point. I'll add a >=

Reply:
Here's how I might rework that fragment to make it a little clearer:
```c
void computeERG(int currentWatts, int setPoint) {
  float incline             = userConfig.getIncline();
  int cad                   = userConfig.getSimulatedCad();
  int newIncline            = incline;
  int amountToChangeIncline = 0;
  const int wattChange      = currentWatts - setPoint;
  const float subShiftScale = 1.0; // Can tweak this value lower as you ride to see what's smoothest
  
  if (cad <= 20) {
    // Cadence too low, nothing to do here
   return;
  }

  amountToChangeIncline = wattChange * userConfig.getERGSensitivity();
  // 30  is amount of watts per shift. Was 50, seemed like too much...
  if (abs(wattChange) < WATTS_PER_SHIFT) {
    amountToChangeIncline *= subShiftScale;
  }
  ...
}
```

Reply:
I'll take that. looks good. I've also reordered the max shift capping...Didn't look right to me. 

Here's what I have currently. 
```void computeERG(int currentWatts, int setPoint) {
  // cooldownTimer--;

  float incline               = userConfig.getIncline();
  int cad                     = userConfig.getSimulatedCad();
  float newIncline            = incline;
  float amountToChangeIncline = 0;
  const int wattChange        = currentWatts - setPoint;
  const float subShiftScale   = 1.0;  // Can tweak this value lower as you ride to see what's smoothest

  if (cad <= 20) {
    // Cadence too low, nothing to do here
    return;
  }
  amountToChangeIncline = wattChange * userConfig.getERGSensitivity();
  // 30  is amount of watts per shift. Was 50, seemed like too much...
  if (abs(wattChange) < WATTS_PER_SHIFT) {
    // As the desired value gets closer, make smaller changes for a smoother experience
    amountToChangeIncline *= subShiftScale;
  }
  // limit to 10 shifts at a time
  if (abs(amountToChangeIncline) > userConfig.getShiftStep() * 10) {
    if (amountToChangeIncline > 0) {
      amountToChangeIncline = userConfig.getShiftStep() * 10;
    }
    if (amountToChangeIncline < 0) {
      amountToChangeIncline = -(userConfig.getShiftStep() * 10);
    }
  }
  // Reduce the amount per loop (don't try to oneshot it.) and reduce the movement the higher the watt target is.
  amountToChangeIncline = amountToChangeIncline / ((currentWatts / 100) + .1);  // +.1 to eliminate possible divide by zero.

  newIncline = incline - amountToChangeIncline;  //  }
  userConfig.setIncline(newIncline);
}

Reply:
A couple questions/issues:

1. What is the point of `subShiftScale` if it is set to `1.0` and is not configurable without recompiling? 
2. With a "properly" configured SS2K, the only time `subShiftScale` will come into effect if the requested change is less than 30w, which is a pretty small change to begin with.
3. Comparison of `wattChange` to `WATTS_PER_SHIFT` doesn't account for sensitivity
4. It is a little confusing the for an "increase" the "amountToChangeIncline" is negative until the very end of the function.

Reply:
1) `subShiftScale` was just giving a name to a magic value that had been there in the past of 0.5. I don't think the goal is to make it user configurable, but I didn't write the original code.
2) My understanding is the goal is to make smaller adjustments when the change is smaller so that a 10W difference causes slower adjustments when closer to the target value. Power reporting can jump up and down a bit so a rider won't be necessarily locked into the target power after hitting it and you don't want to overreact to a small wattage change request.
3) I think that's independent of the sensitivity, it's just asking if the change is bigger or smaller than a shift step

Reply:
@bau-sec is correct. The newly termed subShiftScale which only comes into effect close to the target is to subtly adjust close to the target set point. In my recent testing, it wasn't required by the newest tweaks in the function, but it needs to remain because it may become needed again as others test. 

The bane of every ERG users existence is:
* An ERG function which oscillates above and below a set point.
* An ERG function which drastically overshoots targets. 

subShiftScale accounts for the first. 

The second one is accounted for by capping max shifts, not trying to "one shot" the calculation, and decreasing the amount to change as currentWatts increases. It's important for us to not dwell on getting this exact because we need an adjustable one size fits all solution because different bikes have different resistance mechanisms (magnetic vs friction) as well as different methods to actuate those mechanisms (screws vs cantilevered arms).